services/nftStorage.ts

```diff
@@ -0,0 +1,67 @@
+import { $fetch, FetchError } from 'ohmyfetch'
+import { URLS } from '../utils/constants'
+import consola from 'consola'
+
+const BASE_URL = URLS.koda.nftStorage
+
+const nftStorageApi = $fetch.create({
+  baseURL: BASE_URL,
+})
+
+export type PinningKey = {
+  expiry: string
+  token: string
+}
+
+type StorageApiResponse = {
+  ok: boolean
+  value: {
+    cid: string
+    size: number
+    type: string
+    created: Date
+  }
+}
+
+export const pinJson = async (object: Record<string, any>, name: string) => {
+  const { value } = await nftStorageApi<StorageApiResponse>(`pinJson/${name}`, {
+    method: 'POST',
+    body: object,
+  }).catch((error: FetchError) => {
+    throw new Error(
+      `[NFT::STORAGE] Unable to PIN JSON for reasons ${error.data}`
+    )
+  })
+  consola.log('[NFT::STORAGE] Pin Json')
+  return value.cid
+}
+
+export const getKey = async (address: string): Promise<PinningKey> => {
+  const { expiry, token } = await nftStorageApi<PinningKey>(
+    `getKey/${address}`
+  ).catch((error: FetchError) => {
+    throw new Error(
+      `[NFT::STORAGE] Unable to GET KEY for reasons ${error.data}`
+    )
+  })
+  return { expiry, token }
+}
+
+export const pinFileToIPFS = async (file: Blob, _: string): Promise<string> => {
+  const { value } = await nftStorageApi<StorageApiResponse>('/pinFile', {
+    method: 'POST',
+    body: file,
+    headers: {
+      'Content-Type': file.type ? `${file.type}` : '*/*',
+    },
+  }).catch((error: FetchError) => {
+    throw new Error(
+      `[NFT::STORAGE] Unable to PIN File for reasons ${error.data}`
+    )
+  })
+  consola.log('[NFT::STORAGE] Pin File')
+  return value.cid
+}
+
+export default nftStorageApi
+// QmYt2FydonvVMsEqe2q3hvm38WDq21xM8Z5ZSHZw19PwjF;
```

Comment:
this shouldn't be publicly available on github right?

Reply:
I don't really know what this key is about maybe, should we hide it? It doesn't seem to be used

Reply:
yeah let's hide it just to be safe

Reply:
it's the public hash of the file 

https://nftstorage.link/ipfs/QmYt2FydonvVMsEqe2q3hvm38WDq21xM8Z5ZSHZw19PwjF -> 
https://bafybeie4ty6y2jxljdguxcpnbxxzlujiiycqynrov6zuaqayo4dc3kasti.ipfs.nftstorage.link/

Reply:
but doesn't need to be there tho 👀